keys/key_client.go

```diff
@@ -15,21 +15,48 @@
 package keys
 
 import (
+	"bytes"
 	"encoding/hex"
 	"fmt"
 
+	acm "github.com/hyperledger/burrow/account"
 	"github.com/hyperledger/burrow/logging"
 	logging_types "github.com/hyperledger/burrow/logging/types"
+	"github.com/tendermint/go-crypto"
 )
 
 type KeyClient interface {
-	// Sign needs to return the signature bytes for given message to sign
-	// and the address to sign it with.
-	Sign(signBytesString string, signAddress []byte) (signature []byte, err error)
-	// PublicKey needs to return the public key associated with a given address
-	PublicKey(address []byte) (publicKey []byte, err error)
+	// Sign returns the signature bytes for given message signed with the key associated with signAddress
+	Sign(signAddress acm.Address, message []byte) (signature crypto.Signature, err error)
+
+	// PublicKey returns the public key associated with a given address
+	PublicKey(address acm.Address) (publicKey acm.PublicKey, err error)
+
+	// Generate requests that a key be generate within the keys instance and returns the address
+	Generate(keyName string, keyType KeyType) (keyAddress acm.Address, err error)
+
+	// Returns nil if the keys isntance is health, error otherwise
+	HealthCheck() error
+}
+
+// This mirrors "github.com/monax/keys/crypto/KeyType" but since we have no use for the struct here it seems simpler
+// to replicate rather than cop an import
+type KeyType string
+
+func (kt KeyType) String() string {
+	return string(kt)
 }
 
+const (
+	KeyTypeEd25519Ripemd160         KeyType = "ed25519,ripemd160"
+	KeyTypeEd25519Ripemd160sha256           = "ed25519,ripemd160sha256"
+	KeyTypeEd25519Ripemd160sha3             = "ed25519,sha3"
+	KeyTypeSecp256k1Ripemd160               = "secp256k1,ripemd160"
+	KeyTypeSecp256k1Ripemd160sha256         = "secp256k1,ripemd160sha256"
+	KeyTypeSecp256k1Ripemd160sha3           = "secp256k1,sha3"
+	KeyTypeDefault                          = KeyTypeEd25519Ripemd160
```

Comment:
Is support for all of these types actually being added in this commit, or is this preliminary?

Reply:
There is support for this in our 'reference' implementation of keys - I decided to import all these constants here in anticipation of using them at a future date.